src/main.rs

```diff
@@ -148,19 +126,89 @@ pub async fn delete_helper(pg_client: &mut PgClient, items: &[i32], table_name:
         table_name,
         values.join(",")        
     );
-    tx.execute(&query, &[]).await?;
-    tx.commit().await?;
+
+    sqlx::query(&query)
+    .execute(pg_con_pool)
+    .await?;
 
     Ok(())
 }
 
-async fn sync(
-        base_url: &str, 
-        pg_client: &mut PgClient, 
-        type_arg: &str, 
-        batch_size: u32, 
-        page_resource_count: u32
-    ) -> Result<(), anyhow::Error>{
+// get the number of rows of a given pg table
+pub async fn get_num_rows(pg_con_pool: &PgPool, table_name: &str) -> Result<i64,anyhow::Error> {
+    let query = format!("SELECT COUNT(*) FROM {}", table_name);
+    let row = sqlx::query(query.as_str())
+        .fetch_one(pg_con_pool)
+        .await?;
+    let count: i64 = row.try_get(0)?;
+    Ok(count)
+}
+
+pub async fn get_blaze_search_set(url: &str) -> Result<SearchSet, anyhow::Error> {
+    let client = Client::new();
```

Comment:
Usually we create a http client once and reuse it. Either storing it in a global (easy with `once_cell::sync::Lazy`) or passing it to the functions directly (Its also cheap to clone).